src/lib/storage.ts

```diff
@@ -53,51 +33,63 @@ export class Storage {
     await this.closeDB(db)
   }
 
-  async addPersona(args: {did: string, controllingKey: string}) {
+  async addPersona(args: PersonaOptions) : Promise<void> {
     const db = await this.getDbInstance()
     const query = dbHelper.addPersonaQuery(args)
     await this.executeWriteQuery(db, query)
     await this.closeDB(db)
   }
 
-  async addMasterKey(entropy: string) {
+  async addMasterKey(entropy: string) : Promise<void> {
     const db = await this.getDbInstance()
     const query = dbHelper.addMasterKeyQuery(entropy)
     await this.executeWriteQuery(db, query)
     await this.closeDB(db)
   }
 
-  async addDerivedKey(args: { encryptedWif: string, path: string, entropySource: string, keyType: string }) {
+  async addDerivedKey(args: DerivedKeyOptions) : Promise<void> {
     const db = await this.getDbInstance()
     const query = dbHelper.addDerivedKeyQuery(args)
     await this.executeWriteQuery(db, query)
     await this.closeDB(db)
   }
 
-  async getPersonas() {
+  async getPersonas() : Promise<PersonaOptions[]> {
+    interface ExtendedRowList extends ResultSetRowList {
+      raw: () => PersonaOptions[]
+    }
+
     const db = await this.getDbInstance()
-    const query = 'select * from Keys'
-    const res = await this.executeReadQueryAlt(db, query)
-    return res.rows.raw()
+    const query = dbHelper.getPersonasQuery()
+    const rows = await this.executeReadQuery(db, query) as ExtendedRowList
+    await this.closeDB(db)
+    return rows.raw()
   }
 
   private async createTable(options : TableOptions, db: SQLiteDatabase) : Promise<void> {
     const query = dbHelper.createTableQuery(options)
     await this.executeWriteQuery(db, query)
   }
 
-  private async executeReadQueryAlt(db: SQLiteDatabase, query: string) : Promise<ResultSet> {
-    return new Promise<ResultSet>((resolve, reject) =>
-      db.transaction(async (tx: AsyncTransaction) => {
-        const result = await tx.executeSql(query, [])
-        return resolve(result[1])
-      })
+  private async executeReadQuery(db: SQLiteDatabase, query: AssembledQuery) : Promise<ResultSetRowList> {
+    const { text, values } = query
+
+    return new Promise<ResultSetRowList>((resolve, reject) =>
+      db.readTransaction((tx: Transaction) => {
+        return tx.executeSql(text, values, (tx: Transaction, result: ResultSet) => {
+          return resolve(result.rows)
+        }, reject)
+      }, reject)
     )
   }
 
-  private async executeWriteQuery(db: SQLiteDatabase, query: string) : Promise<void> {
-    return await db.transaction(async (tx: AsyncTransaction) => 
-      tx.executeSql(query, [])
+  private async executeWriteQuery(db: SQLiteDatabase, query: AssembledQuery) : Promise<void> {
```

Comment:
I would explicitly use `Transaction` here as well, as it's not the default way of handling SQL queries.